transformer_engine/pytorch/attention/dot_product_attention/context_parallel.py

```diff
@@ -3485,6 +3485,19 @@ def attn_forward_func_with_cp(
 ) -> torch.Tensor:
     """
     Attention implementation with context parallelism.
+
+    .. note::
+
+        Context parallelism distributes chunks of the sequence onto different GPUs. To help with
+        load balancing, users are expected to reorder their tokens before entering this function.
+        For example, given cp_size = 2, we divide each sequence in a batch into 4 chunks, and
+        distribute chunk 0 and chunk 3 onto GPU 0, and chunk 1 and chunk 2 onto GPU 1. This requires
```

Comment:
I would say there should be some small example of what the end result here should be. For example, this note does not really tell me that those chunks need to be parts of a single tensor, laid out one after another in memory.